mutt.h

```diff
@@ -656,10 +656,26 @@ typedef struct spam_list_t
   struct spam_list_t *next;
 } SPAM_LIST;
 
-#define mutt_new_list() safe_calloc (1, sizeof (LIST))
-#define mutt_new_heap() safe_calloc(1, sizeof(HEAP))
-#define mutt_new_rx_list() safe_calloc (1, sizeof (RX_LIST))
-#define mutt_new_spam_list() safe_calloc (1, sizeof (SPAM_LIST))
+inline LIST *mutt_new_list()
```

Comment:
I think the idiomatic way is to mark the functions `static inline` to avoid unused functions warnings, but maybe `inline` is enough for that?